src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -28,9 +28,11 @@ class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate
 
   override def beforeVerify(input: Program): Program = {
     val allPredIds = input.predicates.collect{ case p if p.body.nonEmpty => p.name }.toSet
-    val recursivePredIds = (checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)).map(_.name)
-    val nonrecursivePredIds = allPredIds.diff(recursivePredIds)
-    val cond = { pred: String => nonrecursivePredIds(pred) }
+    val recursivePreds = checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)
+    val recursivePredIds = recursivePreds.map(_.name)
+    val predsIdsCalledByOtherPreds = recursivePreds.flatMap(nonRecursivePredsCalledBy(_, input, recursivePredIds)).flatten
+    val nonRecursivePredIds = allPredIds.diff(recursivePredIds)
+    val cond = { pred: String => nonRecursivePredIds(pred) && !predsIdsCalledByOtherPreds(pred) }
```

Comment:
Now that I see the `cond` being expanded, I'm thinking maybe it should just be `!recursivePredIds(pred) && !predIdsCalledByRecursivePreds(pred)`. Then we wouldn't have to create and mentally reason about a `diff`. Could you change that for me?